tests/system/src/test/java/test/javafx/scene/web/StraightLineTest.java

```diff
@@ -162,15 +162,18 @@ public void setupTestObjects() {
             // buttom start x position of underline ( startx + font size + thickness -1)
             int line_start_x = start_x + height + 20 - 1;
             // buttom start y position of underline ( startx + height)
-            int line_start_y = start_y + height;
-            String line_color = "rgba(0,0,0,255)"; // color of line
-            for (int i = line_start_y; i < snapshot.getHeight(); i++) {
-                String expected = colorToString(pr.getColor(line_start_x, i));
-                if(expected.equals(line_color))
+            int line_start_y = start_y + height + 3;
+            String expected_line_color = "rgba(0,0,0,255)"; // color of line
+            for (int i = line_start_y; i <= (width - line_start_y -6); i++) {
+                String actual_line_color = colorToString(pr.getColor(line_start_x, i));
+                if (actual_line_color.equals(expected_line_color)) {
                     continue;
-                else
-                    fail("Each pixel color of line should be" + line_color + " but was:" + expected);
+                }
+                else {
```

Comment:
Minor: this should be on one line, like this:

```
    } else {
```

Alternatively, you can replace the entire if-then-else block with:

```
    assertEquals("Pixel color does not match", expected_line_color, actual_line_color);
```